mlrun/serving/states.py

```diff
@@ -1419,6 +1425,24 @@ async def predict_async(
             )
         return body
 
+    def init(self):
+        super().init()
+
+        unchanged_predict = (
+            type(self).predict is LLModel.predict
+            and self._execution_mechanism != storey.ParallelExecutionMechanisms.asyncio
+        )
+        unchanged_async_predict = (
+            type(self).predict_async is LLModel.predict_async
+            and self._execution_mechanism == storey.ParallelExecutionMechanisms.asyncio
+        )
+        if not self.model_provider and (unchanged_predict or unchanged_async_predict):
+            predict_function_name = "predict" if unchanged_predict else "predict_async"
+            raise mlrun.errors.MLRunRuntimeError(
+                f"Model provider could not be determined for model '{self.name}',"
+                f" and the {predict_function_name} function was not overridden."
+            )
```

Comment:
I think this option is more self-explanatory. What do you think?
```suggestion
        if not self.model_provider:
            if self._execution_mechanism != storey.ParallelExecutionMechanisms.asyncio:
                unchanged_predict = self.predict is LLModel.predict
                predict_function_name = "predict"
            else:
                unchanged_predict = self.predict_async is LLModel.predict_async
                predict_function_name = "predict_async"

            if unchanged_predict:
                raise mlrun.errors.MLRunRuntimeError(
                    f"Model provider could not be determined for model '{self.name}',"
                    f" and the {predict_function_name} function was not overridden."
                )
```

Reply:
`self.predict is LLModel.predict` will not work because python binds the method to the instance, I changed it to
`self.__class__.predict is LLModel.predict` that is more elegant than `type(self)`